tools/qgepmaptooladdfeature.py

```diff
@@ -267,6 +267,8 @@ def rightClicked(self, _):
                     f.setAttribute(idx, self.layer.dataProvider().defaultValue(idx))
 
             f.setGeometry(self.rubberband.asGeometry3D())
+            print(f.geometry().isGeosValid())
```

Comment:
`print`s are dangerous (if they happen to be done in a slot). Use QgsMessageLog or the `logging` module instead